lib/livebook/storage/ets.ex

```diff
@@ -98,13 +110,90 @@ defmodule Livebook.Storage.Ets do
 
     :ets.insert(table, attributes)
 
-    {:reply, :ok, state}
+    {:reply, :ok, schedule_persist(state)}
   end
 
   @impl GenServer
   def handle_call({:delete, namespace, entity_id}, _from, %{table: table} = state) do
     :ets.delete(table, {namespace, entity_id})
 
-    {:reply, :ok, state}
+    {:reply, :ok, schedule_persist(state)}
+  end
+
+  @impl GenServer
+  def handle_info(:persist, state) do
+    state = %{state | pending_persist: nil}
+
+    {:noreply, save_to_file(state)}
+  end
+
+  defp schedule_persist(%{pending_persist: ref} = state) do
+    unless is_nil(ref) do
+      Process.cancel_timer(ref)
+    end
+
+    ref = Process.send_after(self(), :persist, @persistence_timeout)
+
+    %{state | pending_persist: ref}
+  end
+
+  defp default_secret() do
+    Livebook.Config.secret!("LIVEBOOK_SECRET_KEY_BASE")
+  end
+
+  defp default_file_path() do
+    # NOTE: should we add the application port to distinguish between multiple instances?
+    Path.join([:filename.basedir(:user_cache, "livebook"), "livebook.conf"])
```

Comment:
That's a good question. I feel like the user would expect the config to apply to all ports, the only tricky scenario is when there are multiple instances at the same time, in which case they would override each other :thinking:

Reply:
Also, we should make the path configurable, for now just application env would be fine. For example, in dev/test we should instead use something like `"tmp/livebook.#{Mix.env()}.conf"`, so it doesn't collide with the global one. @josevalim iirc you were saying we should expect a whole directory like `LIVEBOOK_CACHE`?

Reply:
@josevalim we should add `config :livebook, :data_path, "tmp/livebook_data_test"` to `config/test.exs`, and for dev similarly, right?

Reply:
@jonatanklosko I think that's a good idea, yes! Can you please add this to #1003? We can handle it there!

Reply:
Surething!

Reply:
I believe that it will scream that the `/tmp/livebook_data_test` does not exist (unless we create it ourselves).

Reply:
To be clear, I mean `tmp` in the project dir, but you're right and the creation is also a part of #1003 :D